File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/SchedulePreference.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import java.time.LocalTime;

public class SchedulePreference {
    private Context mContext;
    private SharedPreferences mPreference;
    private final String mKey = "enabledSchedule";

    public static SchedulePreference
    create(Context context){
        return new SchedulePreference(context);
    }

    private SchedulePreference(Context context){
        mContext = context;
        mPreference = PreferenceManager.getDefaultSharedPreferences(mContext.getApplicationContext());
    }
    public void save(boolean enabled){
        SharedPreferences.Editor e = mPreference.edit();
        e.putBoolean(mKey, enabled);
        e.apply();
    }

    public boolean isEnabled(){
        return mPreference.getBoolean(mKey, false);
    }
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/SchedulePreference.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

class SchedulePreference private constructor(private val context: Context) {
    private val preference: SharedPreferences =
        PreferenceManager.getDefaultSharedPreferences(context.applicationContext)
    private val key = "enabledSchedule"
    fun save(enabled: Boolean) {
        val e = preference.edit()
        e.putBoolean(key, enabled)
        e.apply()
    }

    val isEnabled: Boolean
        get() = preference.getBoolean(key, false)

    companion object {
        fun create(context: Context): SchedulePreference {
            return SchedulePreference(context)
        }
    }
}
